src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPoster.java

```diff
@@ -0,0 +1,53 @@
+package com.atlassian.bitbucket.jenkins.internal.status;
+
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketClientFactoryProvider;
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketCredentials;
+import com.atlassian.bitbucket.jenkins.internal.client.exception.BitbucketClientException;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketPluginConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.credentials.BitbucketCredentialsAdaptor;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketBuildStatus;
+import hudson.model.AbstractBuild;
+import hudson.model.TaskListener;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Singleton
+public class BuildStatusPoster {
+
+    private static final Logger LOGGER = Logger.getLogger(BuildStatusPoster.class.getName());
+    @Inject
+    BitbucketClientFactoryProvider bitbucketClientFactoryProvider;
+    @Inject
+    private BitbucketPluginConfiguration pluginConfiguration;
+
+    public void postBuildStatus(AbstractBuild build, TaskListener listener) {
+        BitbucketRevisionAction revisionAction = build.getAction(BitbucketRevisionAction.class);
+        if (revisionAction == null) {
+            return;
+        }
+        Optional<BitbucketServerConfiguration> serverOptional = pluginConfiguration.getServerById(revisionAction.getServerId());
+        if (serverOptional.isPresent()) {
+            BitbucketServerConfiguration server = serverOptional.get();
+            listener.getLogger().println("Posting build status to: " + server.getServerName());
```

Comment:
we may want to include the status here? I don't know. I went for basics first. But happy to have something richer.

Reply:
I'm flexible. Given the feature is inconsequential to the build itself I'm actually a bit hesitant to include too much detail in the logs, but the enum status seems like a reasonable compromise?

Reply:
yeah, I'm happy either way. I just want something in the _build_ logs to show that we attempted to post the build status. The enum seems easy enough, and we have to get hold of it anyway.